sdk/samples/iot/iot_sample_common.c

```diff
@@ -63,12 +63,22 @@ static az_span const provisioning_global_endpoint
 //
 // Functions
 //
-void build_error_message(char* out_full_message, char const* const error_message, ...)
+void build_error_message(
+    char* out_full_message,
+    size_t message_buf_size,
+    char const* const error_message,
+    ...)
 {
   char const* const append_message = ": az_result return code 0x%08x.";
 
-  strcpy(out_full_message, error_message);
-  strcat(out_full_message, append_message);
+  size_t message_len = strlen(error_message) + 1;
+  strncpy(out_full_message, error_message, message_buf_size);
+  out_full_message[message_buf_size - 1] = 0;
```

Comment:
`full_message_buf_size` instead of `message_buf_size`  might be easier to read and follow.
same with `error_message_len` instead of `message_len`

```
  size_t error_message_len = strlen(error_message) + 1;
  strncpy(out_full_message, error_message, full_message_buf_size);
  out_full_message[full_message_buf_size - 1] = 0;
  if (full_message_buf_size > error_message_len)
  {
    strncat(out_full_message, append_message, full_message_buf_size - error_message_len);
    out_full_message[full_message_buf_size - 1] = 0;
  }
}
```